haskell/cc.bzl

```diff
@@ -45,40 +44,21 @@ def cc_interop_info(ctx):
     Returns:
       CcInteropInfo: Information needed for CC interop.
     """
-    hdrs = depset()
-
-    # XXX There's gotta be a better way to test the presence of
-    # CcSkylarkApiProvider.
-    ccs = [dep.cc for dep in ctx.attr.deps if hasattr(dep, "cc")]
-
-    hdrs = depset(transitive = [cc.transitive_headers for cc in ccs])
-
-    hdrs = depset(transitive = [hdrs] + [
-        # XXX cc_import doesn't produce a cc field, so we emulate it with a
-        # custom provider.
-        dep[CcSkylarkApiProviderHacked].transitive_headers
-        for dep in ctx.attr.deps
-        if CcSkylarkApiProviderHacked in dep
-    ])
+    ccs = [dep[CcInfo] for dep in ctx.attr.deps if CcInfo in dep]
 
+    hdrs = depset(transitive = [cc.compilation_context.headers for cc in ccs])
     include_directories = set.to_list(set.from_list(
-        [f for cc in ccs for f in cc.include_directories] +
-        [
-            f
-            for dep in ctx.attr.deps
-            if CcSkylarkApiProviderHacked in dep
-            for f in dep[CcSkylarkApiProviderHacked].include_directories
-        ],
+        [f for cc in ccs for f in cc.compilation_context.includes]
     ))
     quote_include_directories = set.to_list(set.from_list(
-        [f for cc in ccs for f in cc.quote_include_directories],
+        [f for cc in ccs for f in cc.compilation_context.quote_includes],
     ))
     system_include_directories = set.to_list(set.from_list(
-        [f for cc in ccs for f in cc.system_include_directories],
+        [f for cc in ccs for f in cc.compilation_context.system_includes],
     ))
 
     cpp_flags = (
-        ["-D" + define for cc in ccs for define in cc.defines] +
+        ["-D" + define for cc in ccs for define in cc.compilation_context.defines] +
```

Comment:
We walk through the `ccs` list multiple times in the lines above, which is might be a hit on analysis speed.

Reply:
Good point. I've changed it to a single iteration.